agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java

```diff
@@ -1094,7 +1095,80 @@ private static void validateRegex(String value, ProcessorType processorType) {
 
   public static class ProcessorAttribute {
     public String key;
-    public String value;
+    public Object value;
+    public AttributeType type =
+        AttributeType.STRING; // default to string for backward compatibility
+
+    // TODO (heya) remove this and reuse the standalone exporter Mappings.convertToString, need to
+    // make it public
+    public AttributeKey<?> getAttributeKey() {
+      switch (type) {
+        case STRING:
+          return AttributeKey.stringKey(key);
+        case BOOLEAN:
+          return AttributeKey.booleanKey(key);
+        case LONG:
+          return AttributeKey.longKey(key);
+        case DOUBLE:
+          return AttributeKey.doubleKey(key);
+        case STRING_ARRAY:
+          return AttributeKey.stringArrayKey(key);
+        case BOOLEAN_ARRAY:
+          return AttributeKey.booleanArrayKey(key);
+        case LONG_ARRAY:
+          return AttributeKey.longArrayKey(key);
+        case DOUBLE_ARRAY:
+          return AttributeKey.doubleArrayKey(key);
+      }
+      throw new IllegalStateException("Unexpected attribute key type: " + type);
+    }
+
+    public Object getAttributeValue() {
+      if (value instanceof Integer) {
+        if (type == AttributeType.LONG) {
+          return ((Integer) value).longValue();
+        } else if (type == AttributeType.DOUBLE) {
+          return ((Integer) value).doubleValue();
+        }
+      }
+      return value;
+    }
+
+    public void validate() {
+      if (type == AttributeType.STRING && !(value instanceof String)) {
+        throw new FriendlyException(
+            "Attribute type is String, but value is not a string.",
+            "Please provide a valid String attribute value.");
```

Comment:
a bit more info would be helpful for users to pinpoint what the problem in their json file is, e.g.

```suggestion
            "Telemetry processor attribute <name> has type is String, but the value is not a string.",
            "Please provide a json value which matches the attribute type.");
```